File: listing_builder/frontend/src/lib/api/catalog-health.ts
// frontend/src/lib/api/catalog-health.ts
// Purpose: API calls for Catalog Health Check module
// NOT for: React hooks or UI logic (those are in hooks/useCatalogHealth.ts)

import { apiRequest } from './client'

export interface CatalogScan {
  id: string
  user_id: string
  marketplace: string
  seller_id: string | null
  status: 'pending' | 'scanning' | 'completed' | 'failed'
  progress: { phase: string; percent: number } | null
  total_listings: number
  issues_found: number
  issues_fixed: number
  error_message: string | null
  started_at: string | null
  completed_at: string | null
  created_at: string | null
}

export interface CatalogIssue {
  id: string
  scan_id: string
  asin: string | null
  sku: string | null
  issue_type: string
  severity: 'critical' | 'warning' | 'info'
  title: string
  description: string | null
  amazon_issue_code: string | null
  fix_proposal: Record<string, unknown> | null
  fix_status: 'pending' | 'applied' | 'failed' | 'skipped'
  fix_result: Record<string, unknown> | null
  created_at: string | null
}

export interface CatalogDashboard {
  total_scans: number
  last_scan: CatalogScan | null
  issues_by_type: Record<string, number>
  issues_by_severity: Record<string, number>
  total_issues: number
  total_fixed: number
}

export interface FixResult {
  issue_id: string
  fix_status: string
  fix_result: Record<string, unknown> | null
  message: string
}

export interface CatalogHealthStatus {
  credentials_configured: boolean
  has_refresh_token: boolean
}

export async function fetchCatalogHealthStatus(): Promise<CatalogHealthStatus> {
  const res = await apiRequest<CatalogHealthStatus>('get', '/catalog-health/status')
  if (res.error) throw new Error(res.error)
  return res.data!
}

export async function fetchCatalogDashboard(): Promise<CatalogDashboard> {
  const res = await apiRequest<CatalogDashboard>('get', '/catalog-health/dashboard')
  if (res.error) throw new Error(res.error)
  return res.data!
}

export async function startCatalogScan(marketplace: string): Promise<CatalogScan> {
  const res = await apiRequest<CatalogScan>('post', '/catalog-health/scan', { marketplace })
  if (res.error) throw new Error(res.error)
  return res.data!
}

export async function fetchScanStatus(scanId: string): Promise<CatalogScan> {
  const res = await apiRequest<CatalogScan>('get', `/catalog-health/scan/${scanId}`)
  if (res.error) throw new Error(res.error)
  return res.data!
}

export async function fetchScanIssues(
  scanId: string,
  params?: { issue_type?: string; severity?: string; offset?: number; limit?: number }
): Promise<{ issues: CatalogIssue[]; total: number }> {
  const res = await apiRequest<{ issues: CatalogIssue[]; total: number }>(
    'get', `/catalog-health/scan/${scanId}/issues`, undefined, params as Record<string, unknown>
  )
  if (res.error) throw new Error(res.error)
  return res.data!
}

export async function fetchScanHistory(
  params?: { offset?: number; limit?: number }
): Promise<{ scans: CatalogScan[]; total: number }> {
  const res = await apiRequest<{ scans: CatalogScan[]; total: number }>(
    'get', '/catalog-health/scans', undefined, params as Record<string, unknown>
  )
  if (res.error) throw new Error(res.error)
  return res.data!
}

export async function applyIssueFix(issueId: string): Promise<FixResult> {
  const res = await apiRequest<FixResult>('post', `/catalog-health/fix/${issueId}`)
  if (res.error) throw new Error(res.error)
  return res.data!
}

File: listing_builder/backend/schemas/catalog_health.py
# backend/schemas/catalog_health.py
# Purpose: Pydantic schemas for Catalog Health Check API
# NOT for: Database models (models/catalog_health.py)

from datetime import datetime
from typing import Any, Dict, List, Literal, Optional

from pydantic import BaseModel, Field

ALLOWED_MARKETPLACES = ("DE", "FR", "IT", "ES", "PL", "NL", "SE", "BE", "UK", "US")


class ScanStartRequest(BaseModel):
    marketplace: Literal["DE", "FR", "IT", "ES", "PL", "NL", "SE", "BE", "UK", "US"] = Field(
        default="DE", description="Amazon marketplace code"
    )


class ScanResponse(BaseModel):
    id: str
    user_id: str
    marketplace: str
    seller_id: Optional[str] = None
    status: str
    progress: Optional[Dict[str, Any]] = None
    total_listings: int = 0
    issues_found: int = 0
    issues_fixed: int = 0
    error_message: Optional[str] = None
    started_at: Optional[datetime] = None
    completed_at: Optional[datetime] = None
    created_at: Optional[datetime] = None

    class Config:
        from_attributes = True


class ScanListResponse(BaseModel):
    scans: List[ScanResponse]
    total: int


class CatalogIssueResponse(BaseModel):
    id: str
    scan_id: str
    asin: Optional[str] = None
    sku: Optional[str] = None
    issue_type: str
    severity: str
    title: str
    description: Optional[str] = None
    amazon_issue_code: Optional[str] = None
    fix_proposal: Optional[Dict[str, Any]] = None
    fix_status: str = "pending"
    fix_result: Optional[Dict[str, Any]] = None
    created_at: Optional[datetime] = None

    class Config:
        from_attributes = True


class IssueListResponse(BaseModel):
    issues: List[CatalogIssueResponse]
    total: int


class DashboardResponse(BaseModel):
    total_scans: int = 0
    last_scan: Optional[ScanResponse] = None
    issues_by_type: Dict[str, int] = {}
    issues_by_severity: Dict[str, int] = {}
    total_issues: int = 0
    total_fixed: int = 0


class FixResultResponse(BaseModel):
    issue_id: str
    fix_status: str
    fix_result: Optional[Dict[str, Any]] = None
    message: str
